### ISSUE TITLE
How to prevent trailing slash being added to the paths in additionalPaths

### ISSUE THREAD

**Issue by dominicarrojado** [iamvishnusankar/next-sitemap#258]
**Describe the bug**
A trailing slash is being added to the paths in additionalPaths. I [previously created a bug report](https://github.com/iamvishnusankar/next-sitemap/issues/223) but it was closed, assuming it was fixed, I tried with the latest version but still experience the same behaviour so recreating the issue again.

**To Reproduce**

1. Update `next-sitemap.js`:
```
module.exports = {
  siteUrl: 'https://example.com',
  generateRobotsTxt: true,
  additionalPaths: async (config) => [
    await config.transform(config, '/additional-page.html'),
  ],
};

```
2. Build and export next app
3. Result:
```
<url><loc>https://example.com/additional-page.html/</loc><changefreq>daily</changefreq><priority>0.7</priority><lastmod>2021-10-15T07:05:21.440Z</lastmod></url>
```

**Expected behavior**
It shouldn't include trailing slash if not needed - `<loc>https://example.com/additional-page.html</loc>`. When crawlers access the page with trailing slash (e.g. `https://example.com/additional-page.html/`), the page will break because of the assets it requires. It should be accessed by crawlers without the trailing slash.

**Additional context**
I'm using `next-sitemap@1.8.4`.

**Comment by iamvishnusankar** [comment]
@dominicarrojado This is not a bug. You're passing the config with `trailingSlash` to your transform function. By, default `next-sitemap` works based on the `trailingSlash` value of `next.config.js`. If you want to override this, you can pass a `trailingSlash : false` to `next-sitemap.js`. Or if you only want to override this behaviour specific paths, you can do it via transform function as well. (See the example below)

```js
module.exports = {
  siteUrl: 'https://example.com',
  generateRobotsTxt: true,
  trailingSlash: false, // This will override the value read from next.config.js (global)
  additionalPaths: async (config) => [
    await config.transform(
      {
        ...config,
        trailingSlash: false, // Per path override
      },
      '/additional-page.html'
    ),
  ],
}
```

**Comment by dominicarrojado** [comment]
@iamvishnusankar hey thanks for the reply. I actually forgot to mention that I tried that before too, anyway I tried it again with the latest version both of your examples (adding it globally and per path override) didn't have the expected behaviour, it still adds trailingSlash. I've created a sandbox for you to check too - https://codesandbox.io/s/nifty-wood-vyehw?file=/public/sitemap.xml. You will be able to open another tab in the terminal by clicking the `+` sign and run `yarn build`. See that `public/sitemap.xml` produces `<loc>https://example.com/additional-page.html/</loc>` instead of `<loc>https://example.com/additional-page.html</loc>`.

**Comment by vinaynair39** [comment]
@dominicarrojado were you able to find a solution for this?

**Comment by dominicarrojado** [comment]
@vinaynair39 no I didn't find a solution, I'm still waiting for @iamvishnusankar to look into it.

**Comment by mordgard** [comment]
@iamvishnusankar I have vise versa issue with that slash. I need to add it in specific route, e.g 
``` result.push({
      loc: `${config.siteUrl}/${lang}/`,
      ...defaultOptions
    });```
    
    and it generates url like `<url><loc>https://example.com/en</loc><lastmod>2022-02-23T11:21:14.131Z</lastmod></url>` without trailing slash.

**Comment by iamvishnusankar** [comment]
@dominicarrojado @mordgard Fixed #310

**Comment by dominicarrojado** [comment]
@iamvishnusankar I'm afraid it is still not fixed: https://codesandbox.io/s/interesting-tu-toztvx?file=/next.config.js

When there's `next.config.js` with the `trailingSlash: true`. the `next-sitemap.js` config does not override it.
